File: files/2025-06-29/CSEC 29062025.R
library(tidyverse)
library(readxl)

path = "files/2025-06-29/Challenge 37.xlsx"
input = read_excel(path, range = "B3:F10")
test  = read_excel(path, range = "H3:O4") %>% 
  rename(Item = `...1`)

result = input %>%
  mutate(Col = coalesce(Level, Gender, Location)) %>%
  select(-c(Level, Gender, Location)) %>%
  pivot_wider(names_from = Col, values_from = Values) 

all.equal(result, test)
# > [1] TRUE

File: files/2025-06-29/CSEC 29062025.py
import pandas as pd

path = "files/2025-06-29/Challenge 37.xlsx"

input = pd.read_excel(path, usecols="B:F", skiprows=2, nrows=8)
test = pd.read_excel(path, usecols="H:O", skiprows=2, nrows=1)
test = test.rename(columns={test.columns[0]: "Item"})

input['Col'] = input[['Level', 'Gender', 'Location']].bfill(axis=1).iloc[:, 0]
result = input.pivot(index='Item', columns='Col', values='Values').reset_index()[test.columns]

print(result.equals(test)) # True
